dist/api_data.py

```diff
@@ -1760,13 +1760,16 @@ def __ge__(self, other):
         timestamp_roundup''',
         type='category', subconfig= [
         Config('prepared', 'false', r'''
-            applicable only for prepared transactions. Indicates if the prepare
-            timestamp and the commit timestamp of this transaction can be
-            rounded up. If the prepare timestamp is less than the stable
-            timestamp, the prepare timestamp will be rounded to the stable
-            timestamp. If the commit timestamp is less than the prepare
-            timestamp, the commit timestamp will be rounded up to the prepare
-            timestamp''', type='boolean'),
+            applicable only for prepared transactions, and intended only for special-purpose
+            use. (See @ref timestamp_roundup_prepare.) Allows the prepare timestamp and the
+            commit timestamp of this transaction to be rounded up to be no older than the
+            oldest timestamp, and allows violating the usual restriction that the prepare
+            timestamp must be newer than the stable timestamp. Specifically: if the prepare
+            timestamp is less than or equal to the oldest timestamp, the prepare timestamp
+            will be rounded to the oldest timestamp. Then, if the commit timestamp is less
+            than the prepare timestamp, the commit timestamp will be rounded up to the prepare
```

Comment:
Apologies if I'm being pedantic -- but I was asked to look at a documentation change. Effective communication to the most impacted audience is an artful skill that I've been unable to hone.

In the following, I'd advise replacing `prepare` with `oldest`:
```
Then, if the commit timestamp is less than the prepare timestamp, the commit timestamp will be rounded up to the prepare timestamp.
```

Specifically:
```
Then, if the commit timestamp is also less than the oldest timestamp, the commit timestamp will be also be rounded up to the oldest timestamp.
```

The statement is using the context that the (original) prepared value has been rounded. And the line as written is referring to the "rounded" prepared value. I first read that as the commit timestamp being rounded to the "original" prepared value. Knew it couldn't be true, then corrected the parsing. But I don't think many readers of this will have the confidence to declare my original interpretation wrong and instead choose to be silently confused.

Reply:
Yes, I'd wondered if that was clear enough and eventually decided it was, but evidently was wrong :-)

The problem with your suggested wording is that I think it's technically wrong: the commit timestamp is rounded up to the updated prepare timestamp without reference to oldest, which could in theory have changed. Changing it thus would be wrong, of course... how about "Subsequently, at commit time, if the commit timestamp is less than the (now rounded) prepare timestamp, the commit timestamp will be rounded up to it and thus to at least oldest." And I'll add a note in the main page that it's expected that oldest doesn't change during application recovery.

Reply:
Explicitly referring to the rounded prepared timestamp works for me.